llvm/include/llvm/CodeGen/MachineSSAUpdater.h

```diff
@@ -40,8 +40,8 @@ class MachineSSAUpdater {
   //typedef DenseMap<MachineBasicBlock*, Register> AvailableValsTy;
   void *AV = nullptr;
 
-  /// VRC - Register class of the current virtual register.
-  const TargetRegisterClass *VRC = nullptr;
+  /// RegAttrs - current virtual register, new registers copy its attributes.
+  Register RegAttrs;
```

Comment:
Relying on an existing register (which may mutate?) may be cumbersome. Maybe would be better to store the RegClassOrRegBank + LLT?

Reply:
Proposed changes are in https://github.com/llvm/llvm-project/pull/73337, had to update amdgpu to do the same